src/cluster/cluster.go

```diff
@@ -1,90 +1,86 @@
 package cluster
 
 import (
-	"firecontroller/microcontroller"
-	mc "firecontroller/microcontroller"
-	"firecontroller/utilities"
+	device "devicecommander/device"
+	"errors"
 	"log"
+	"time"
 
 	"github.com/spf13/viper"
 )
 
-//Cluster - This object defines an array of microcontrollers
+//Cluster - This object defines an array of Devices
 type Cluster struct {
-	Name             string
-	Microcontrollers []mc.Microcontroller
-	Me               *mc.Microcontroller
+	Name    string
+	Devices []device.Device
 }
 
-//Config -
-type Config struct {
-	Name             string `yaml:"Name"`
-	Microcontrollers []mc.Config
-}
-
-//GetConfig -
-func (c Cluster) GetConfig() (config Config) {
-	config.Name = c.Name
-	config.Microcontrollers = make([]mc.Config, len(c.Microcontrollers))
-	for i, micro := range c.Microcontrollers {
-		config.Microcontrollers[i] = micro.GetConfig()
+//GetDevices returns a map of all registered
+func (c Cluster) GetDevices() map[string]device.Device {
+	devices := make(map[string]device.Device)
+	for i := 0; i < len(c.Devices); i++ {
+		devices[c.Devices[i].ID] = c.Devices[i]
 	}
-
-	return
+	return devices
 }
 
-//Master - returns a pointer to the Master micro
-func (c *Cluster) Master() *mc.Microcontroller {
-	for _, micro := range c.Microcontrollers {
-		if micro.Master {
-			return &micro
+//AddDevice attempts to add a device to the cluster and returns the response data.
+func (c *Cluster) AddDevice(newDevice device.Device) error {
+	if viper.GetString("ENV") == "production" {
+		for _, micro := range c.Devices {
+			if micro.Host == newDevice.Host && newDevice.Port == micro.Port {
+				return errors.New("This host & port combination are already registered to this cluster")
```

Comment:
We could add a TODO: here that is something like `search through the device list and prune out duplicate serial registrations.`. Otherwise:
- Scenario: a device changes ip address.
- Risk: the device could take up many slots in the list